File: examples/typescript/manage-webhooks.ts
/**
 * Create, update, and delete a webhook on a phone number.
 *
 * Usage:
 *   INKBOX_API_KEY=ApiKey_... PHONE_NUMBER_ID=<id> npx ts-node manage-webhooks.ts
 */

import { InkboxPhone } from "../../typescript/src/phone/index.js";

const client = new InkboxPhone({ apiKey: process.env.INKBOX_API_KEY! });
const phoneNumberId = process.env.PHONE_NUMBER_ID!;

// Create
const hook = await client.webhooks.create(phoneNumberId, {
  url: "https://example.com/webhook",
  eventTypes: ["incoming_call"],
});
console.log(`Created webhook ${hook.id}  secret=${hook.secret}`);

// Update
const updated = await client.webhooks.update(phoneNumberId, hook.id, {
  url: "https://example.com/webhook-v2",
});
console.log(`Updated URL: ${updated.url}`);

// Delete
await client.webhooks.delete(phoneNumberId, hook.id);
console.log("Deleted.");

File: examples/python/manage_webhooks.py
"""
Create, update, and delete a webhook on a phone number.

Usage:
    INKBOX_API_KEY=sk-... PHONE_NUMBER_ID=<id> python manage_webhooks.py
"""

import os
from inkbox.phone import InkboxPhone

client = InkboxPhone(api_key=os.environ["INKBOX_API_KEY"])
phone_number_id = os.environ["PHONE_NUMBER_ID"]

# Create
hook = client.webhooks.create(
    phone_number_id,
    url="https://example.com/webhook",
    event_types=["incoming_call"],
)
print(f"Created webhook {hook.id}  secret={hook.secret}")

# Update
updated = client.webhooks.update(
    phone_number_id,
    hook.id,
    url="https://example.com/webhook-v2",
)
print(f"Updated URL: {updated.url}")

# Delete
client.webhooks.delete(phone_number_id, hook.id)
print("Deleted.")
